### PR TITLE
fix(safe-apps): Safe apps analysis improvements

### PR DESCRIPTION
**Pull request by yagopv**
## What it solves

Add some fixes:

- Duplicated "Open Safe App" event tracking
- 404 when retrieving the apps default list and no chainId in redux store
- Remove querystrings when storing url's from custom apps in analytics (We detect eth addresses there)

### CODE REVIEW

**Review by DaniSomoza — APPROVED** [review]

File: src/services/analytics/gtm.ts [hunk 1]
```diff
@@ -141,13 +141,15 @@ export const gtmTrackPageview = (pagePath: string): void => {
   gtmSend(gtmEvent)
 }
 
+const stripQueryStringAndHashFromPath = (url: string) => url.split('?')[0].split('#')[0]
```
Comment (katspaugh) [inline src/services/analytics/gtm.ts:144]:
```suggestion
const normalizeAppName = (appName?: string): string => {
  // App name is a URL
  if (appName?.startsWith('http')) {
    // Strip search query and hash
    return appName.split('?')[0].split('#')[0]
  }
  return appName || ''
}
```

Worth adding some tests?

Comment (yagopv) [reply src/services/analytics/gtm.ts]: done

**Review by katspaugh — APPROVED** [review]
👍

**Comment by liliya-soroka** [comment]
Verified

**Pull request merged**